### PR TITLE
Fix HLint warnings

### PR DESCRIPTION
**Pull request by cblp**

### CODE REVIEW

File: src/Ormolu/Exception.hs [hunk 1]
```diff
@@ -84,7 +86,7 @@ withPrettyOrmoluExceptions m = m `catch` h
           OrmoluParsingFailed _ _ -> 3
           OrmoluOutputParsingFailed _ _ -> 4
           OrmoluASTDiffers _ _ -> 5
-          OrmoluNonIdempotentOutput _ _ _ -> 6
```
Comment (neongreen) [inline src/Ormolu/Exception.hs]: Can you change the other branches as well?

File: src/Ormolu/Printer/Meat/ImportExport.hs [hunk 2]
```diff
@@ -36,9 +36,7 @@ p_hsmodImport ImportDecl {..} = do
   space
   when ideclQualified (txt "qualified")
   space
-  case ideclPkgQual of
-    Nothing -> return ()
-    Just slit -> atom slit
+  forM_ ideclPkgQual atom
```
Comment (neongreen) [inline src/Ormolu/Printer/Meat/ImportExport.hs]: Meh. I'd argue this one should just be disabled in `.hlint.yaml`.

File: src/Ormolu/Printer/Meat/Type.hs [hunk 3]
```diff
@@ -145,7 +146,7 @@ p_hsType' multilineArgs = \case
   where
     isPromoted = \case
       HsTyVar _ IsPromoted _ -> True
-      HsExplicitListTy _ _ _ -> True
+      HsExplicitListTy {} -> True
       HsExplicitTupleTy _ _ -> True
```
Comment (neongreen) [inline src/Ormolu/Printer/Meat/Type.hs]: `HsExplicitTupleTy` should also be changed then.

File: src/Ormolu/Exception.hs (hunk 1 above)
Comment (cblp) [reply src/Ormolu/Exception.hs]: Done

File: src/Ormolu/Printer/Meat/ImportExport.hs (hunk 2 above)
Comment (cblp) [reply src/Ormolu/Printer/Meat/ImportExport.hs]: Reverted.

Should `.hlint.yaml` be added?

File: src/Ormolu/Printer/Meat/Type.hs (hunk 3 above)
Comment (cblp) [reply src/Ormolu/Printer/Meat/Type.hs]: Done

File: src/Ormolu/Printer/Meat/ImportExport.hs (hunk 2 above)
Comment (neongreen) [reply src/Ormolu/Printer/Meat/ImportExport.hs]: I would add it, yeah. Once I start using HLint, I usually end up needing a `.hlint.yaml` almost immediately.

Comment (cblp) [reply src/Ormolu/Printer/Meat/ImportExport.hs]: If this should be done, this should be done a separate PR.

Comment (mrkkrp) [reply src/Ormolu/Printer/Meat/ImportExport.hs]: `.hlint.yaml` should be added if we decide to run Hlint on CI. Otherwise no.

**Pull request merged**